src/testing/match/match.go

```diff
@@ -35,40 +37,46 @@ func main() {
 	}
 	re := regexp.MustCompile(flag.Args()[0])
 
-	// Read through input looking for match
-	s := bufio.NewScanner(os.Stdin)
-	exit := -1
+	// Read through input looking for match && copying input to inputBuf (which is
+	// written to stdout at the end)
+	inputBuf := &strings.Builder{}
+	matchFound := false
+	s := bufio.NewScanner(io.TeeReader(os.Stdin, inputBuf))
 	for s.Scan() {
-		os.Stdout.Write(s.Bytes())
-		os.Stdout.Write([]byte{'\n'})
-		if exit >= 0 {
-			continue
-		}
 		switch {
-		case !re.Match(s.Bytes()): // No match, continue
-		case *invert: // Failure -- matched with -v
+		case matchFound:
+			continue // match was found in the past--just copy input to inputBuf
+		case !re.Match(s.Bytes()):
+			continue // no match--keep copying and looking
+		case *invert:
+			// Failure: matched with -v
 			fmt.Fprintf(
 				os.Stderr,
-				"did not expect to find\n  \"%s\"\nbut matched\n  \"%s\"\nin the text:\n",
+				"match for forbidden regex\n  \"%s\"\nin the text:\n  \"%s\"\n",
 				flag.Args()[0],
 				s.Text(),
 			)
-			exit = 1
-		default: // Success, match -- copy input to output
-			exit = 0
+			os.Exit(1)
+		default:
+			matchFound = true // match found on this line
 		}
 	}
-	if exit >= 0 {
+	if matchFound || *invert {
```

Comment:
not a big deal, but i feel like what `invert` is actually doing is very straightforward relative to the somewhat confusing if or else if not logic here

Reply:
I agree that it's a little confusing. I tried to clarify w/ the comment in the if body, but if you see a better way I'm happy to go with it.

Reply:
alright, i think this is good! -- i tried to adjust it but i couldn't think of a cleaner way